yb-voyager/cmd/live_migration.go

```diff
@@ -412,28 +411,36 @@ func handleEvent(event *tgtdb.Event, evChans []chan *tgtdb.Event) error {
 }
 
 // Returns a hash value between 0..NUM_EVENT_CHANNELS
-func hashEvent(e *tgtdb.Event) int {
+func hashEvent(e *tgtdb.Event, tableToPartitioningStrategyMap *utils.StructMap[sqlname.NameTuple, string]) (int, error) {
 	hash := fnv.New64a()
-	hash.Write([]byte(e.TableNameTup.ForKey()))
-	if tableToPartitioningStrategyMap != nil {
-		if strategy, ok := tableToPartitioningStrategyMap.Get(e.TableNameTup); ok && strategy == PARTITION_BY_TABLE {
-			return int(hash.Sum64() % (uint64(NUM_EVENT_CHANNELS)))
-		}
+
+	if tableToPartitioningStrategyMap == nil {
+		return 0, fmt.Errorf("table to partitioning strategy map is not initialized")
 	}
-	//else if tableToPartitioningStrategyMap is nil or the strategy is not PARTITION_BY_PK, then use the PARTITION_BY_PK strategy
 
-	//include key columns in the hash
-	keyColumns := make([]string, 0)
-	for k := range e.Key {
-		keyColumns = append(keyColumns, k)
+	strategy, ok := tableToPartitioningStrategyMap.Get(e.TableNameTup)
+	if !ok {
+		return 0, fmt.Errorf("table to partitioning strategy map does not contain table %v", e.TableNameTup)
 	}
 
-	// sort to ensure input to hash is consistent.
-	sort.Strings(keyColumns)
-	for _, k := range keyColumns {
-		hash.Write([]byte(*e.Key[k]))
+	switch strategy {
+	case PARTITION_BY_PK:
+		hash.Write([]byte(e.TableNameTup.ForKey()))
+		//include key columns in the hash
+		keyColumns := make([]string, 0)
+		for k := range e.Key {
+			keyColumns = append(keyColumns, k)
+		}
+
+		// sort to ensure input to hash is consistent.
+		sort.Strings(keyColumns)
+		for _, k := range keyColumns {
+			hash.Write([]byte(*e.Key[k]))
+		}
+	case PARTITION_BY_TABLE:
```

Comment:
we should add a case default and return error. 
Wherever we do not expect any other case, it's always better to error out. Otherwise, it might silently just pass through...